tests/integration/test_migrator.py

```diff
@@ -142,7 +164,41 @@ def test_continuous_migration_handles_multiple_records(app, record_1502655_and_1
     assert expected == result
 
 
+def test_continuous_migration_avoids_jamming(app, many_times_same_record):
+    with app.app_context():
+        with mock.patch.dict(app.config, {'CELERY_ALWAYS_EAGER': False}):
```

Comment:
I had a similar issue but I didn't used mocks. The solution was to create the `app` with the `CELERY_ALWAYS_EAGER` flag.

```
app = create_app(
        DEBUG=True,
        WTF_CSRF_ENABLED=False,
        CELERY_ALWAYS_EAGER=True,
        CELERY_RESULT_BACKEND='cache',
        CELERY_CACHE_BACKEND='memory',
        CELERY_EAGER_PROPAGATES_EXCEPTIONS=True,
        TESTING=True,
    )
```